engine/runtime/src/main/java/org/enso/interpreter/node/callable/dispatch/InvokeFunctionNode.java

```diff
@@ -108,15 +108,20 @@ Object invokeCached(
     if (cachedSchema.getCallerFrameAccess().shouldFrameBePassed()) {
       callerInfo = captureCallerInfoNode.execute(callerFrame.materialize());
     }
-    functionCallInstrumentationNode.execute(
-        callerFrame, function, state, mappedArguments.getSortedArguments());
-    return curryNode.execute(
-        callerFrame,
-        function,
-        callerInfo,
-        state,
-        mappedArguments.getSortedArguments(),
-        mappedArguments.getOversaturatedArguments());
+    var result =
+        functionCallInstrumentationNode.execute(
+            callerFrame, function, state, mappedArguments.getSortedArguments());
+    if (result instanceof FunctionCallInstrumentationNode.FunctionCall) {
```

Comment:
It seemed to me that there is a support for `on_call` to return _different value_ than the function would compute otherwise. I wrote a test in f65c19a - however then I realized the changed value (result of multiplication) is ignored here.

It is easy to change that and use the value, when provided. Do we want to do it? I like it as it is in the  [spirit of GraalVM Insight](https://www.graalvm.org/latest/tools/graalvm-insight/manual/#intercepting-and-altering-execution). However we don't need it in Enso right now and I can also change the documentation of `on_call` to say that the result of callback is ignored.

Please vote! +1 to allow altering execution flow & result, -1 to ignore result of `on_call` callback.

Reply:
+1